var/ramble/repos/builtin/modifiers/wait-for-bg-jobs/modifier.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2022-2024 The Ramble Authors
+#
+# Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+# https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+# <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+# option. This file may not be copied, modified, or distributed
+# except according to those terms.
+
+from ramble.modkit import *
+import ramble.config
+from ramble.util.executable import CommandExecutable
+from ramble.util.shell_vars import last_pid_var
+
+
+class WaitForBgJobs(BasicModifier):
+    """Define a modifier for waiting for background jobs.
+
+    This only waits for background jobs started from the experiment.
```

Comment:
This only works for background jobs in bash, right? If so, I think that's fine, but we should add a comment in the docstring that this only works for bash just so people know. We could also make the builtins / modifications conditional on the shell being bash (since I don't think sh has arrays, but I could be wrong on that)

Reply:
That's a good catch, indeed this seems to be bash only. (Other newer shells like fish has array support, but the syntax seems different.) Added the check (instead of skipping it will error out when other shells are detected.)